numpy/core/src/private/get_attr_string.h

```diff
@@ -1,37 +1,55 @@
 #ifndef __GET_ATTR_STRING_H
 #define __GET_ATTR_STRING_H
 
-static NPY_INLINE int
-_is_basic_python_type(PyObject * obj)
+static NPY_INLINE npy_bool
+_is_basic_python_type(PyTypeObject *tp)
 {
-    if (obj == Py_None ||
-            PyBool_Check(obj) ||
-            /* Basic number types */
+    PyTypeObject const * const known_types[] = {
```

Comment:
Can't the compiler optimize the array since its const, and full of addresses to static variables?

I could add a `static` if it helps